### PR TITLE
duplicacy-web init at 1.8.1

### PR DESCRIPTION
**Pull request by DogeRam1500**
## Description of changes

Packaged duplicacy-web, a web-based frontend for the Duplicacy backup program. The homepage can be found at [1](https://duplicacy.com/) and release notes for 1.8.1 can be found at [2](https://forum.duplicacy.com/t/duplicacy-web-edition-1-8-1-releases/8419). It's also available as a dmg file for MacOS, but I haven't been able to figure out how to handle those.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Result of `nixpkgs-review` run on x86_64-linux [3](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>duplicacy-web</li>
  </ul>
</details>

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/du/duplicacy-web/package.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+in
+
+stdenv.mkDerivation {
+  pname = "duplicacy-web";
+  version = version;
+
+  src = fetchurl {
+    url = urls.${stdenv.hostPlatform.system};
+    sha256 = hashes.${stdenv.hostPlatform.system};
+  };
```
Comment (pluiedev) [inline pkgs/by-name/du/duplicacy-web/package.nix:24]: Honestly I would just do something like this
```nix
let 
  inherit (stdenv.hostPlatform) system;

  platforms = {
    x86_64-linux = {
      url = "...";
      hash = "...";
    };
    #...
  };
in
stdenv.mkDerivation {
  #...
  src = fetchurl (platforms.${system} or throw "Unsupported system: ${system}");
  #...
  meta = {
    platforms = lib.attrNames platforms;
    #...
  };
}
```

File: pkgs/by-name/du/duplicacy-web/package.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  phases = [ "installPhase" ];
```
Comment (pluiedev) [inline pkgs/by-name/du/duplicacy-web/package.nix:26]: Never set `phases` directly. Use the `dontUnpack`, `dontBuild`, etc flags instead

File: pkgs/by-name/du/duplicacy-web/package.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp $src $out/bin/duplicacy-web
+    chmod +x $out/bin/duplicacy-web
+  '';
+
+  meta = with lib; {
```
Comment (pluiedev) [inline pkgs/by-name/du/duplicacy-web/package.nix:34]: [Avoid `with lib;`](https://github.com/NixOS/nixpkgs/issues/208242)

File: pkgs/by-name/du/duplicacy-web/package.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    homepage = "https://duplicacy.com";
+    description = "A new generation cloud backup tool with web-based GUI";
+    platforms = platforms.linux;
+    license = licenses.unfree;
+    maintainers = [ maintainers.DogeRam1500 ];
+    downloadPage = "https://duplicacy.com/download.html";
+  };
```
Comment (pluiedev) [inline pkgs/by-name/du/duplicacy-web/package.nix:41]: Unfree binaries sourced from built native code must have its source provenance set
```suggestion
    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
  };
```

**Review by pluiedev — CHANGES_REQUESTED** [review]

File: pkgs/by-name/du/duplicacy-web/package.nix [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  hashes = {
+    "x86_64-linux" = "sha256-XgyMlA7rN4YU6bXWP52/9K2LhEigjzgD2xQenGU6dn4=";
+    "aarch64-linux" = "sha256-M2RluQKsP1002khAXwWcrTMeBu8sHgV8d9iYRMw3Zbc=";
+    "armv5tel-linux" = "sha256-O4CHtKiRTciqKehwCNOJciD8wP40cL95n+Qg/NhVSGQ=";
+  };
+in
+
+stdenv.mkDerivation {
+  pname = "duplicacy-web";
+  version = version;
```
Comment (pluiedev) [inline pkgs/by-name/du/duplicacy-web/package.nix:19]: Just use `inherit version;`. Alternatively, use `finalAttrs` to make this deriv override-friendly

**Comment by DogeRam1500** [comment]
I think I'm going to close this, since I clearly need to spend more time learning how to use Nix. Maybe I'll try again some time in the future.

**Comment by pluiedev** [comment]
> I think I'm going to close this, since I clearly need to spend more time learning how to use Nix. Maybe I'll try again some time in the future.

I mean, this is the kind of routine scrutiny that every PR faces in Nixpkgs — please don't take it personally 😅

**Pull request closed without merging**